app/src/main/kotlin/com/wire/android/util/FileUtil.kt

```diff
@@ -39,16 +45,50 @@ suspend fun Uri.toByteArray(context: Context): ByteArray {
     }
 }
 
-fun getWritableImageAttachment(context: Context) = getTempWritableAttachmentUri(context, TEMP_IMG_ATTACHMENT_FILENAME)
-
-fun getWritableVideoAttachment(context: Context) = getTempWritableAttachmentUri(context, TEMP_VIDEO_ATTACHMENT_FILENAME)
+fun Context.getTempWritableImageUri() = getTempWritableAttachmentUri(this, TEMP_IMG_ATTACHMENT_FILENAME)
+fun Context.getTempWritableVideoUri() = getTempWritableAttachmentUri(this, TEMP_VIDEO_ATTACHMENT_FILENAME)
+fun Context.getDownloadsFolderUri(assetName: String) = getTempWritableAttachmentUri(this, assetName)
 
 private fun getTempWritableAttachmentUri(context: Context, fileName: String): Uri {
     val file = File(context.cacheDir, fileName)
     file.setWritable(true)
     return FileProvider.getUriForFile(context, BuildConfig.APPLICATION_ID + ".provider", file)
 }
 
+fun Context.saveFileDataToDownloadsFolder(downloadedFile: File, fileSize: Int): Uri? {
```

Comment:
this all is going to be executed on the MainThread I suppose

Reply:
I think working on files and with buffers would be a good canditate to extract it to a seperate class that would be injectable inside the ViewModel, that way all those processes will survive configuration changes and could be exectued on for example IO thread's

Reply:
commented above